cfspopcon/formulas/fusion_power/fusion_power_reduction.py

```diff
@@ -0,0 +1,35 @@
+"""Reduce fusion power to limit by decreasing heavier fuel species fraction."""
+
+import numpy as np
+import xarray as xr
+from numpy import float64
+from numpy.typing import NDArray
+
+from ...algorithm_class import Algorithm
+from ...unit_handling import Unitfull
+
+
+@Algorithm.register_algorithm(return_keys=["heavier_fuel_species_fraction"])
+def require_P_fusion_less_than_P_fusion_limit(
+    P_fusion_upper_limit: Unitfull,
+    P_fusion: Unitfull,
+    heavier_fuel_species_fraction: NDArray[float64],
+) -> Unitfull:
+    """Change heavier_fuel_species_fraction to reduce P_fusion to P_fusion_limit.
+
```

Comment:
```suggestion
    """Change heavier_fuel_species_fraction to reduce P_fusion to P_fusion_limit.
    
    Derived by solving
    Pfus = fT0 (1 - fT0) nD kfus
    Pmax = fT (1 - fT) nD kfus
    for fT (new_heavier_fuel_species_fraction), where fT0 is the original heavier_fuel_species_fraction, using the
    quadratic formula.
```
N.b. this formula won't work with DD fusion. Probably don't need to worry about this since we'll be miles away from Pmax if doing DD fusion, but might be worth adding
```
if isinstance(fusion_reaction, xr.DataArray):
    fusion_reaction = fusion_reaction.item()
reaction = REACTIONS[fusion_reaction]
if not isinstance(reaction, (DTFusionBoschHale, DTFusionHively)):
    raise NotImplementedError(f"Cannot reduce fusion power for reaction {reaction} by changing fuel ratio")

```